types/wlr_idle_inhibit_v1.c

```diff
@@ -0,0 +1,191 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <wlr/util/log.h>
+#include <wlr/types/wlr_surface.h>
+#include <wlr/types/wlr_idle_inhibit_v1.h>
+#include "wayland-util.h"
+#include "wayland-server.h"
+#include "idle-inhibit-unstable-v1-protocol.h"
+
+struct wlr_idle_inhibit_manager {
+	struct wlr_idle_inhibit_v1 *wlr_idle_inhibit;
+	struct wl_resource *resource;
+
+	struct wl_list link; // wlr_idle_inhibit_v1::clients
+
+	struct wl_list inhibitors;
+};
+
+static void idle_inhibit_inhibitor_destroy(struct wl_resource *resource) {
+	struct wlr_idle_inhibit_inhibitor_v1 *inhibitor = wl_resource_get_user_data(resource);
```

Comment:
Wrap `wl_resource_get_user_data` with a helper function that checks the type of the resource.